ikarus/solver/nonlinearsolver/newtonraphson.hh

```diff
@@ -176,12 +177,12 @@ public:
         linearSolver_.factorize(Ax);
         linearSolver_.solve(correction_, -rx);
         dNorm = correction_.norm();
-        updateFunction_(x, correction_);
       } else {
         correction_ = -linearSolver_(rx, Ax);
         dNorm       = norm(correction_);
-        updateFunction_(x, correction_);
       }
+      updateStates(nonLinearOperator().assembler(), correction_);
```

Comment:
I dont like that introduction of states inside the innocent Newton-Raphson function.
This entangels unrelated functionality and might drop on our feet later, sicne extensibility is compromised since a local change somewhere else might affect this here.